packages/scheduler-prisma/src/helpers.ts

```diff
@@ -12,34 +12,44 @@ import {
 } from './index';
 import { Properties } from './types';
 
+const SIX_MONTHS_IN_MS = 15778476000;
+
 export const isHealthy = (heartbeat: number, timeout: number) =>
   new Date().getTime() - timeout < heartbeat;
 
 // interval should be iCal String.
 export const computeNextRunAt = (
   interval: string,
   timezone = 'UTC'
-): moment.Moment => {
+): string => {
   if (!interval) {
     throw new Error('Invalid interval argument supplied to computeNextRunAt');
   }
 
-  const rule = Rule.parse(interval);
+  const isValidRule = interval.includes('DTSTART');
 
-  if (!rule.tzId) {
-    rule.tzId = timezone;
-  }
+  if (!isValidRule) {
+    const rule = interval
+      .split(';')
+      .filter(b => !b.includes('TZID'))
+      .join(';');
 
-  if (!rule.byMinute) {
-    rule.byMinute = [0];
+    const timeISO8601 = moment().tz(timezone).format('YYYYMMDDTHHmmss');
+    const rrule = RRuleSet.parse(
+      `DTSTART;TZID=${timezone}:${timeISO8601}\nRRULE:${rule}\nEXDATE;TZID=${timezone}:${timeISO8601}`
+    );
+    return new Date(rrule.all(1)[0]).toISOString();
   }
 
-  if (!rule.bySecond) {
-    rule.bySecond = [0];
-  }
+  const rrule = RRuleSet.parse(interval);
 
-  const rrule = new Rule(rule);
-  return rrule.getNext(new Date()).date;
+  return new Date(
+    rrule.between(
+      new Date().getTime(),
+      new Date().getTime() + SIX_MONTHS_IN_MS,
```

Comment:
@johndagostino @rrsmith Added the support for a valid Ical rrule.
The caveat here is the until date that I have hard coded to 6 months so we don't generate infinitely which I think is a good compromise. Just means our UI needs to be aware of this limit.